generic3g/specs/FrequencyAspect.F90

```diff
@@ -170,50 +183,40 @@ end function supports_conversion_general
    logical function supports_conversion_specific(src, dst) result(supports)
       class(FrequencyAspect), intent(in) :: src
       class(StateItemAspect), intent(in) :: dst
+      integer :: status
+      logical :: compatible
 
       select type(dst)
       class is (FrequencyAspect)
-         supports = src .divides. dst
+         call check_compatibility(dst, src, compatible, rc=status)
+         supports = compatible .and. status == _SUCCESS
```

Comment:
Yeah - it was unfortunate that the first cases I looked at for this could not return a failure.   It meant that the interface for `...specific` did not need a return code.  

Probably the right thing to do is to modify the interface throughout the subclasses to have a return code.  But **NOT** as part of this PR.    Should be a separate PR.

Reply:
Once you do that though, it it might obviate the need to have `check_compatibility` as that is just a thin wrapper on a procedure that you could just call directly here.